packages/shared-metrics/src/dependencies.js

```diff
@@ -269,3 +310,83 @@ function sortByDistance(distances, dependency1, dependency2) {
   }
   return d2 - d1;
 }
+
+/**
+ * Either reads the contents of the main package.json file or returns mainPackageJsonContents if it was cached already
+ * @param {string} mainPackageJsonPath
+ * @param {(err: Error, contents: string) => void} cb
+ */
+function getMainPackageJsonContents(mainPackageJsonPath, cb) {
+  if (mainPackageJsonContents) {
+    cb(null, mainPackageJsonContents);
+  } else {
+    fs.readFile(mainPackageJsonPath, { encoding: 'utf8' }, (readFileErr, contents) => {
+      if (readFileErr) {
+        logger.debug(`Error reading ${mainPackageJsonPath}`, readFileErr);
+        cb(readFileErr, null);
+      } else {
+        mainPackageJsonContents = contents;
+        cb(null, mainPackageJsonContents);
+      }
+    });
+  }
+}
+
+/**
+ * @param {string} mainPackageJsonPath
+ * @param {Object.<string, *>} packages
+ */
+function detectInstalledDevDependencies(mainPackageJsonPath, packages) {
+  if (downstreamConnection) {
```

Comment:
Could you please add a note here that serverless is using shared metrics package but downstreamConnection will be null and that's why it will be skipped. Thanks :)